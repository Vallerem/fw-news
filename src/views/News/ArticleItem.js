import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Card, CardImgOverlay, CardText, CardTitle, CardImg } from "reactstrap";
import { trimContent } from "../../utils/stringTrimmer";

const ArticleItem = props => {
  const { article } = props;

  return (
    <div className="col-12 col-md-6 col-lg-4 col-xl-4">
      <Card
        className="article-card no-gutters"
        inverse
        tag={Link}
        to={`/${article.id}`}
      >
        <CardImg
          style={{ height: 320 }}
          width="100%"
          src={article.image}
          alt={`article about ${article.title}`}
        />
        <CardImgOverlay style={{ background: "rgba(0,0,0,0.4)" }}>
          <CardTitle>{article.title}</CardTitle>
          <CardText>{trimContent(article.content, 30)}</CardText>
          <CardText style={{ position: "absolute", bottom: 20, left: 20 }}>
            {article.date}
          </CardText>
        </CardImgOverlay>
      </Card>
    </div>
  );
};

ArticleItem.propTypes = {
  article: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
    source: PropTypes.string,
    content: PropTypes.string,
    date: PropTypes.string
  }).isRequired
};

export default ArticleItem;
