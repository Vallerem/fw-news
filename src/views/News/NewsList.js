import React, { Component } from "react";
import { connect } from "react-redux";

import newsAPI from "../../api/news";
import { logout } from "../../redux/actions/user";

import ArticleItem from "./ArticleItem";

export class NewsList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      articles: []
    };
  }

  componentDidMount = () => {
    this.getNews()
      .then(data => {
        this.setState({
          articles: data.news
        });
      })
      .catch(err => this.props.logout());
  };

  getNews = () => {
    return newsAPI.getNews();
  };

  render() {
    const { articles } = this.state;
    return (
      <div className="row mt-4 mb-4 no-gutter">
        {articles.map((article, index) => (
          <ArticleItem key={index} article={article} />
        ))}
      </div>
    );
  }
}

export default connect(null, { logout })(NewsList);
