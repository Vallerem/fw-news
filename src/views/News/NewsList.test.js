import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { NewsList } from "./NewsList";
import ArticleItem from "./ArticleItem";
import sinon from "sinon";

const wrapper = mount(<NewsList logout={() => {}} />);

describe("<NewsList />", () => {
  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("checks the state", () => {
    expect(NewsList.prototype, "componentDidMount").toEqual({});
  });
});
