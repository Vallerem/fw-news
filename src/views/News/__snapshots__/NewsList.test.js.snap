// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsList /> renders correctly 1`] = `
<NewsList
  logout={[Function]}
>
  <div
    className="row mt-4 mb-4 no-gutter"
  />
</NewsList>
`;
